test/Confluent.Kafka.IntegrationTests/Tests/OnPartitionsAssignedNotSet.cs

```diff
@@ -0,0 +1,79 @@
+// Copyright 2016-2017 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+using System;
+using System.Text;
+using System.Collections.Generic;
+using Confluent.Kafka.Serialization;
+using Xunit;
+
+
+namespace Confluent.Kafka.IntegrationTests
+{
+    /// <summary>
+    ///     Test that Assign gets called automatically in a scenario where a
+    ///     handler has not been added to OnPartitionsAssigned
+    ///     (deserializing Consumer)
+    /// </summary>
+    public static partial class Tests
+    {
+        [Theory, MemberData(nameof(KafkaParameters))]
+        public static void OnPartitionsAssignedNotSet(string bootstrapServers, string topic)
```

Comment:
I was going to comment that it this test seems a roundabout way to test the assigned partitions instead of using a simple accessor, but I discover that librdkafka doesn't seem to provide the functionality of checking assigned partitions (unless I'm just stupidly overlooking it)! We might want to consider adding that since it's state that's annoying to have to track yourself and could be handy during pause/resume partition operations.